### PR TITLE
Update notifications

### PR DESCRIPTION
**Pull request by anuveyatsu**
Implemented module for notifying users about new versions of the cli tool:

- 'lib/utils/update.js' is the module which is responsible for update notifications. It will check for updates with caching time of 1h using 'update-notifier' library and will print instructions for updating via NPM as well as updating as an executable binary. Instructions differ depending on OS.

Example for macos:

screen shot 2018-02-12 at 15 35 23

### CODE REVIEW

**Review by Mikanebu — APPROVED** [review]
Good!

**Comment by Mikanebu** [comment]
@anuveyatsu This is a different issue, but we need to add some instructions for `windows` users here http://datahub.io/docs/getting-started/installing-data, wdyt?

**Comment by anuveyatsu** [comment]
@Mikanebu yes, we definitely need it since we have the windows version of the CLI. Can you please create an issue about it?

**Comment by Mikanebu** [comment]
@anuveyatsy created an issue https://github.com/datahq/datahub-qa/issues/110

File: lib/utils/update.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    pkg,
+    updateCheckInterval: 1000 * 60 * 60
+  })
+
+  if (!notifier.update) {
+    return
+  }
+
+  // Depending on running OS show appropriate instructions:
+  let instructions = 'If you\'ve installed data tool using our executable binary then follow instructions below:\n'
```
Comment (AcckiyGerman) [inline lib/utils/update.js]: I don't like hardcoding the instructions like
```javascript
instructions = "..."
if (...) instructions += "..."
if (...) instructions += "..."
```

The code looks not good and It is hard to maintain instructions in the future.

Could you extract instructions in the 
```javascript
instructions = {
  intro: "...",
  for_linux : "...",
  for_win : "...",
  for_mac: "..."
}
```
structure (could be in the separate file) and use it in the conditions like `if (...) console.log(instructions.for_win)`

**Update**
instruction object keys could be `process.platform` and then you can use it like `print(instructions[process.platform]`

Comment (AcckiyGerman) [reply lib/utils/update.js]: @anuveyatsu

Comment (anuveyatsu) [reply lib/utils/update.js]: @AcckiyGerman ok, thanks, think it is a good suggestion!

**Pull request merged**